manifests/init.pp

```diff
@@ -25,6 +25,12 @@
   if ( $facts['os']['family'] == 'RedHat' and $facts['os']['release']['major'] == '8' ){
     fail('You cannot use the cd4pe module to install on EL 8')
   }
+
+  $compiling_server_osfamily = cd4pe::compiling_server_osfamily()
+  if ( $compiling_server_osfamily != $facts['os']['family']){
```

Comment:
From the previous PR feedback, please also restrict on os version being the same.

I saw your comment saying that you somehow got ubuntu 16.04 to work from an 18 master, however that is not expected to work, and I wouldn't want to support that.